frontend/packages/console-dynamic-plugin-sdk/src/api/component-api-types.ts

```diff
@@ -0,0 +1,16 @@
+import { RouteComponentProps, match } from 'react-router';
+import { K8sResourceCommon } from '../extensions';
+
+/* Horizontal Nav Types */
+export type NavPage = {
+  href?: string;
+  path?: string;
+  name: string;
+  component: React.ComponentType<RouteComponentProps>;
+};
+
+export type HorizontalNavProps = {
+  resource?: K8sResourceCommon;
+  pages: NavPage[];
+  match: match;
```

Comment:
Does it really make sense to pass through `match` as a prop?
`HorizontalNav` can use `withRouter` to request the `match` property itself instead of putting the burden on each consumer to do the same.

Reply:
Using  `withRouter` in the facade component to make the SDK user's life easier.